cmake/DynamicVersion.cmake

```
@@ -429,13 +439,13 @@ function(get_dynamic_version)
 				OUTPUT_STRIP_TRAILING_WHITESPACE
 				COMMAND_ERROR_IS_FATAL ANY)
 		# Get version and describe-name
-		execute_process(COMMAND ${GIT_EXECUTABLE} describe --tags --match=?[0-9.]*
+		execute_process(COMMAND ${GIT_EXECUTABLE} describe --tags --long --abbrev=8 --match=?[0-9.]*
```

```suggestion
		execute_process(COMMAND ${GIT_EXECUTABLE} describe --tags --long --match=?[0-9.]*
```

I just realized, `setuptools_scm` uses a 7 character short hash by default (prefixed by `g`), we should also align with that by default.

---

[This](https://github.com/git/git/blob/aeddbfdfa48443c034a9b28b10dfddf2f71b907f/Documentation/RelNotes/2.11.0.txt#L22) is a bit old, but I don't think `git describe` can be relied upon for any particular number of digits. And users can be running really old git, too. That's mostly fyi as you're storing the whole sha anyways. The capture here only saves some cmake manipulations. Glad to remove the arg if you don't want to deal with sha length here.

---

Yeah,  personally think 8 character sha would be better, but since setuptools-scm uses the [default describe](https://setuptools-scm.readthedocs.io/en/latest/config/#setuptools_scm.git.DEFAULT_DESCRIBE), it would be better to align to that